src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -98,6 +98,15 @@ public static StringComparer Create(CultureInfo culture, bool ignoreCase)
             return new CultureAwareComparer(culture, ignoreCase);
         }
 
+        public static StringComparer Create(CultureInfo culture, CompareOptions options)
```

Comment:
If we introduce this new API I would expect the existing one `static StringComparer StringComparer.Create(CultureInfo, bool)` to be marked as obsolete.

Reply:
@danmosemsft should I mark this as obsolete ?

Reply:
No. it is still ok to use

Reply:
We do not mark new API with the obsolete attribute, because historically it makes it hard for customers with large codebases and "warnings as errors" to upgrade their platforms. Instead you could open an issue in https://github.com/dotnet/platform-compat to flag it dynamically.

Reply:
Yes I wasn't talking about `ObsoleteAttribute` but about the compat tooling.